### DISCUSSION TITLE
Video stuttering when running yt videos from qb with mpv

### DISCUSSION THREAD

**Discussion by erichamers** [qutebrowser/qutebrowser#6907] [General] [answered]
When I try to open youtube videos using mpv from qutebrowser I can't get a large enough cache to actually watch the video, it keeps freezing. But when I do it from the command line it actually works fine. Any ideas on why this happens? 

My bind on `config.py`:
`config.bind('M', 'hint links spawn mpv {hint-url}')`

**Comment by The-Compiler** [comment]
No clue... do you use some kind of special environment or sandboxing for qutebrowser (Flatpak, Firejail, AppArmor, etc.)? Could you check via `ps` or so that the command-line being run is indeed the same?

**Comment by bitraid** [comment, marked as answer]
If you're using youtube-dl, try switching to yt-dlp. See https://github.com/ytdl-org/youtube-dl/issues/29326 and https://github.com/yt-dlp/yt-dlp/pull/1437.

**Reply by erichamers** [reply]
This actually solved it.

Had to install `yd-dlp (sudo pacman -S yd-dlp)` and add the following line to `mpv.conf`:

`script-opts=ytdl_hook-ytdl_path={path_to_yt-dlp_binary}`

Thanks for the help.
